### PR TITLE
Integrate with Gradle Enterprise plugin 3.0

### PR DESCRIPTION
**Pull request by ldaley**
CI: https://builds.gradle.org/project.html?projectId=Gradle&tab=projectOverview&branch_Gradle=ldaley%2Fenterprise-plugin

### CODE REVIEW

File: subprojects/core/src/main/java/org/gradle/internal/scan/config/BuildScanPluginCompatibility.java [hunk 1]
```diff
@@ -18,13 +18,14 @@
...
-    private static final String MIN_SUPPORTED_VERSION_DISPLAY = "2.0.2";
-    public static final String UNSUPPORTED_PLUGIN_VERSION_MESSAGE =
-        "This version of Gradle requires version " + MIN_SUPPORTED_VERSION_DISPLAY + " of the build scan plugin or later.\n"
-            + "Please see https://gradle.com/scans/help/gradle-incompatible-plugin-version for more information.";
+    public static final String FIRST_GRADLE_ENTERPRISE_PLUGIN_VERSION_DISPLAY = "3.0";
+    public static final VersionNumber FIRST_GRADLE_ENTERPRISE_PLUGIN_VERSION = VersionNumber.parse(FIRST_GRADLE_ENTERPRISE_PLUGIN_VERSION_DISPLAY);
+
+    public static final String OLD_SCAN_PLUGIN_VERSION_MESSAGE =
+        "The build scan plugin is not compatible with this version of Gradle.\n"
+            + "Please see https://gradle.com/help/gradle-6-build-scan-plugin for more information.";
```
Comment (guylabs) [inline subprojects/core/src/main/java/org/gradle/internal/scan/config/BuildScanPluginCompatibility.java]: Where will this redirect? To a new section in the build scan plugin manual?

File: subprojects/soak/soak.gradle.kts [hunk 2]
```diff
@@ -45,7 +45,7 @@ gradlebuildJava {
 tasks.integTest {
     options {
         require(this is JUnitOptions)
-        excludeCategories("org.gradle.soak.categories.SoakTest")
+//        excludeCategories("org.gradle.soak.categories.SoakTest")
```
Comment (guylabs) [inline subprojects/soak/soak.gradle.kts]: Can this be removed?

File: subprojects/core/src/main/java/org/gradle/internal/scan/config/BuildScanPluginCompatibility.java (hunk 1 above)
Comment (ldaley) [reply subprojects/core/src/main/java/org/gradle/internal/scan/config/BuildScanPluginCompatibility.java]: We'll need to set up a redirect to a new section, yes.

Comment (guylabs) [reply subprojects/core/src/main/java/org/gradle/internal/scan/config/BuildScanPluginCompatibility.java]: Ok. Added a task to https://github.com/gradle/dotcom/issues/3980

File: gradle/init-scripts/build-scan.init.gradle.kts [hunk 3]
```diff
@@ -1,20 +1,20 @@
...
-    rootProject {
-        pluginManager.withPlugin("com.gradle.build-scan") {
-            extensions["buildScan"].withGroovyBuilder {
-                if (gradle.startParameter.taskNames in tasksWithBuildScansOnFailure) {
-                    "publishOnFailure"()
-                } else {
-                    "publishAlways"()
-                }
-                setProperty("server", "https://e.grdev.net")
+    settingsEvaluated {
```
[the remark this answers is not in the document]
Comment (ldaley) [reply gradle/init-scripts/build-scan.init.gradle.kts]: I have made it compatible with all Gradle versions.

File: subprojects/instant-execution/src/integTest/groovy/org/gradle/instantexecution/AbstractInstantExecutionAndroidIntegrationTest.groovy [hunk 4]
```diff
@@ -86,5 +86,13 @@ abstract class AbstractInstantExecutionAndroidIntegrationTest extends AbstractIn
...
     void copyRemoteProject(String remoteProject) {
         new TestFile(new File("build/$remoteProject")).copyTo(testDirectory)
+        updateScanPlugin()
+    }
+
+    private void updateScanPlugin() {
+        // Plugin is no longer compatible
+        buildFile.text -= 'apply plugin: "com.gradle.build-scan"'
+        buildFile.text -= "id 'com.gradle.build-scan' version '2.1' apply false"
+        settingsFile.text = "plugins { id 'com.gradle.enterprise' version '3.0' }\n\n" + settingsFile.text
```
[the remark this answers is not in the document]
Comment (lptr) [reply subprojects/instant-execution/src/integTest/groovy/org/gradle/instantexecution/AbstractInstantExecutionAndroidIntegrationTest.groovy]: I think your plan makes sense. I always wanted to make the tested app be as close to https://github.com/google/santa-tracker-android as possible. I created https://github.com/gradle/gradle-private/issues/2809 to handle this. We need to merge the code as is for now I guess.

[the remark this answers is not in the document]
Comment (ldaley) [reply subprojects/instant-execution/src/integTest/groovy/org/gradle/instantexecution/AbstractInstantExecutionAndroidIntegrationTest.groovy]: Thanks.

**Comment by ldaley** [comment]
@big-guy should be complete.

**Pull request merged**